### DISCUSSION TITLE
Nuxt 3 Vite sub-optimal loading/bundling strategy

### DISCUSSION THREAD

**Discussion by AndrewBogdanovTSS** [nuxt/nuxt#15916] [Questions]
During the development of Nuxt 3, I noticed how my application is currently loading in production and it looks like every component is loading as a separate tiny chunk which leads to 160 requests on a relatively simple page. Is it intended behavior or I'm missing something? I'm using nuxt/components to register components in the application and for those from small chunks I'm NOT using the lazy prefix
What I think should be intended behavior: 
Components should be bundled based on the criteria of the page they are being used, individual component-related chunks should only be created for the possibility of lazy loading, but those should not be called if the lazy prefix wasn't specified for the component used on the page.

**Comment by danielroe** [comment]
This is rollup's default behaviour (it never repeats code in more than one chunk). You can configure it via `vite.build.rollupOptions.output.manualChunks`.

**Comment by KlausSchaefersAtWork** [comment]
Can you give an example how to solve this?
